### PR TITLE
DOCS-2297 : New sketch of a Mule domain docs page (Mule 4)

### PR DESCRIPTION
**Pull request by ethanport**
I couldn't see Mule domains documented, so here's a start.

### CODE REVIEW

File: mule-user-guide/v/4.0/mule-domain-projects.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+
+
+    <!-- configure here resource to be shared within the domain -->
+
+</domain:mule-domain>
+----
++
+You can copy/paste resources from a Mule application, but you will also have to carefully copy/paste the namesapce declaration and the corresponding XSD files.
```
Comment (marianogonzalez) [inline mule-user-guide/v/4.0/mule-domain-projects.adoc]: you shouldn't have to copy paste anything. Studio is supposed to do that for you

File: mule-user-guide/v/4.0/mule-domain-projects.adoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+For example, here is the dependency you would add to the Mule application's POM file to associate the Mule application with the muledomain2 Mule domain:
+
++
+[source,xml,linenums]
+----
+	<dependencies>
+		<dependency>
+            <groupId>com.mycompany</groupId>
+            <artifactId>muledomain2</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
```
Comment (marianogonzalez) [inline mule-user-guide/v/4.0/mule-domain-projects.adoc]: add a note clarifying that this is for development only. Production applications should never depend on a snapshot, which means that the domain needs to be published before the application does

File: mule-user-guide/v/4.0/mule-domain-projects.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+----
++
+You can copy/paste resources from a Mule application, but you will also have to carefully copy/paste the namesapce declaration and the corresponding XSD files. 
+
+
+Instead, you can use the Global Elements editor to add shared elements to the domain project. 
+
+Just like with a Mule application, you will have to import the module for whatever connector or other module you want to add to the domain project. But because the Mule domain project does not include a flow editor, you instead click the Manage Module button in the Global Elements editor. 
+This will bring up a Properties editor. 
+Then click Add Module to add a common module directly from inside Studio, or you can seach for other modules in Anypoint Exchange.
```
Comment (marianogonzalez) [inline mule-user-guide/v/4.0/mule-domain-projects.adoc]: can probably use screenshots here.

**Comment by dukesphere** [comment]
Once approved, seems like this should go with the domain docs in /anypoint-studio/v/7.1 rather than in mule4-user-guide/v/4.1 (note that mule-user-guide/v/4.0 no longer exists).

**Pull request closed without merging**